### ISSUE TITLE
History entries not showing up in chrome://history

### ISSUE THREAD

**Issue by N0body42** [Eloston/ungoogled-chromium#1735]
**Describe the bug**
The browser saves my history (as I want it to), and I can tell by trying to clear it, as shown in the screenshot below. When I go to chrome://history I am met with a blank page that says "Your browsing history appears here". Sites I have visited have purple links and everything but it won't let me view my history for some reason.

**To Reproduce**
1. Go to 'chrome://history' and notice that there are no history entries.
2. Click on 'Clear browsing data'
3. Under 'Browsing history' it says I have approxiamately 3.8k history entries in total. Changing the time range updates the amount of entries, as expected.

**Expected behavior**
I would expect the history entries to show up in chrome://history.

**Screenshots**

![screenshot16Nov21_2](https://user-images.githubusercontent.com/83465851/142007375-d7b77357-8d10-4b09-9698-e73d97c70f8c.png)

![screenshot16Nov21_1](https://user-images.githubusercontent.com/83465851/142007426-705018a2-bdc9-44e3-a511-605e25fe137c.png)

Not sure how to prove it but these are taken pretty much seconds apart, and no, I never cleared this history.

**Environment (please complete the following information):**
 - OS/Platform and version: Pop!_OS 21.04
 - ungoogled-chromium version: 91.0.4472.164 (Official Build, ungoogled-chromium) (64-bit)

**Comment by PF4Public** [comment]
How did you install ungoogled-chromium and from where?

**Comment by N0body42** [comment]
> How did you install ungoogled-chromium and from where?

I downloaded the "Linux AppImage 64-bit" from https://ungoogled-software.github.io/ungoogled-chromium-binaries/

**Comment by PF4Public** [comment]
Try "Portable Linux 64-bit". This could be AppImage issue.

**Comment by kutukvpavel** [comment]
I ran into this issue too on 95.0.4638.69-1 (Windows x64) right after a clean installation on a machine that never had Chromium installed on it. Setting Startup Options to "Start from where you left off" also causes "Something went wrong when opening your profile" error message that was my initial concern. Makes sense, if it can't read history databse.

I used my SQL Server tools to peek inside the History database and it was almost empty. It contained a couple of URLs that seem to be some default values for my region (?, I never visited them after Chromium installation), tables 'visits' and 'keyword_search_terms' were blank. So I deleted the database file, after this chrome://history came back to life.

I'm not a Chromium developer, but some differences between 'corrupted' and 'regenerated' versions of the history database caught my attention, namely the number of tables and the version value in 'meta' table (see screenshots).
![image](https://user-images.githubusercontent.com/41666738/143660814-c0567832-ba41-4a78-944c-3c722cd3c3ee.png)
![image](https://user-images.githubusercontent.com/41666738/143660842-d60d504e-2748-4807-b66f-f74d4d1e3475.png)
I can post full database files if needed.

---

TL;DR: I removed History database file from my Chromium user profile (along with its journal file, but in hindsight I don't think it's necessary) and it fixed the issue for me. To preseve your data you'll probably have to edit the database somehow.

---
P.S. Does an empty history database have to contain some default URLs? Why are they in Russian (a part of Yandex ecosystem)? Does it somehow detect my location during installation?

**Comment by PF4Public** [comment]
> TL;DR: I removed History database file from my Chromium user profile (along with its journal file, but in hindsight I don't think it's necessary) and it fixed the issue for me.

So that this issue didn't reoccur after cleaning the profile?

**Comment by kutukvpavel** [comment]
@PF4Public so far I'm good.

I'm probably going to install chromium on one more Windows machine tomorrow, so I'll probably have a chance to explore this issue some more.

**Comment by PF4Public** [comment]
@kutukvpavel Let us know if you find something interesting.

**Comment by kutukvpavel** [comment]
@PF4Public unfortunately, I was unable to reproduce this issue on the Win10 machine, got frustrated by configuring the same browser settings again and again in different order and eventually gave up on this. I have one other Win7 machine left, I hope I'll try it on weekend.

There is are some differences between the machine I used when I encountered the issue and the ones I used after that. First, the first machine had Google Chrome installed and the others have MS Edge (the chromium-based one) installed. Though I can't imagine how this fact could've influenced Chromium, since it doesn't feature data import from Chrome/Edge (AFAIK). Second, the first machine has regional settings all set to Russia, and the second one has mixed regional settings (AFAIR the UI langugage is changed to English).

My guess is: this was either some sort of a weird edge case that arises when you set up the brower in a certain order after installation, or something about region-dependent defaults.

**Comment by PF4Public** [comment]
@kutukvpavel Thank you for your efforts. I'll keep this issue open for a while then.
